src/autoscaler/metricsforwarder/manager/policyManager.go

```diff
@@ -0,0 +1,123 @@
+package manager
+
+import (
+	"autoscaler/db"
+	"autoscaler/models"
+	"sync"
+	"time"
+
+	"code.cloudfoundry.org/clock"
+	"code.cloudfoundry.org/lager"
+	cache "github.com/patrickmn/go-cache"
+)
+
+type Consumer func(map[string]*models.AppPolicy, chan *models.AppMonitor)
+type GetPoliciesFunc func() map[string]*models.AppPolicy
+
+type PolicyManager struct {
+	logger             lager.Logger
+	interval           time.Duration
+	cacheTTL           time.Duration
+	allowedMetricCache cache.Cache
+	database           db.PolicyDB
+	clock              clock.Clock
+	doneChan           chan bool
+	policyMap          map[string]*models.AppPolicy
+	pLock              sync.RWMutex
+	mLock              sync.RWMutex
+}
+
+func NewPolicyManager(logger lager.Logger, clock clock.Clock, interval time.Duration,
+	database db.PolicyDB, allowedMetricCache cache.Cache, cacheTTL time.Duration) *PolicyManager {
+	return &PolicyManager{
+		logger:             logger.Session("PolicyManager"),
+		clock:              clock,
+		interval:           interval,
+		cacheTTL:           cacheTTL,
+		allowedMetricCache: allowedMetricCache,
+		database:           database,
+		doneChan:           make(chan bool),
+		policyMap:          make(map[string]*models.AppPolicy),
+	}
+}
+func (pm *PolicyManager) GetPolicies() map[string]*models.AppPolicy {
+	pm.pLock.RLock()
+	defer pm.pLock.RUnlock()
+	return pm.policyMap
+}
+func (pm *PolicyManager) Start() {
+	go pm.startPolicyRetrieve()
+	pm.logger.Info("started", lager.Data{"interval": pm.interval})
+}
+
+func (pm *PolicyManager) Stop() {
+	close(pm.doneChan)
+	pm.logger.Info("stopped")
+}
+
+func (pm *PolicyManager) startPolicyRetrieve() {
+	tick := pm.clock.NewTicker(pm.interval)
+	defer tick.Stop()
+
+	for {
+		policyJsons, err := pm.retrievePolicies()
+		if err != nil {
+			continue
+		}
+		policies := pm.computePolicies(policyJsons)
+
+		pm.pLock.Lock()
+		pm.policyMap = policies
+		pm.pLock.Unlock()
+
+		isRefreshed := pm.RefreshAllowedMetricCache(policies)
```

Comment:
looks like isRefreshed will be false if there is error when replacing cache.  I would suggest to 
1) return error instead of true/false, and log the error
2) when there is error,  still wait for next cycle. In the current impl, it keeps query policy db and replace cache, which will cause performance issue.